### PR TITLE
rest api

### PR DESCRIPTION
**Pull request by ochenmudro**

### CODE REVIEW

File: src/main/java/com/example/demo/controller/ClientController.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    private final ClientMapper clientMapper;
+
+    public ClientController(ClientService clientService, ClientMapper clientMapper) {
+        this.clientService = clientService;
+        this.clientMapper = clientMapper;
+    }
+
+    @GetMapping
+    public ResponseEntity<String> getClient(@RequestParam Integer clientId) {
+        if (clientId == null) return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```
Comment (varya1441) [inline src/main/java/com/example/demo/controller/ClientController.java]: add brackets
code is not readable

File: src/main/java/com/example/demo/controller/ClientController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @GetMapping(value = "all")
+    public ResponseEntity<List<ClientDTO>> getAllClients() {
+        List<ClientDTO> clients = this.clientService.getAll().stream().map(clientMapper::mapToDTO).toList();
+        if (clients.isEmpty()) return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        return new ResponseEntity<>(clients, HttpStatus.OK);
+    }
+
+    @PutMapping(value = "attend/")
```
Comment (varya1441) [inline src/main/java/com/example/demo/controller/ClientController.java]: you dont need /

File: src/main/java/com/example/demo/controller/ClientController.java (hunk 1 above)
Comment (varya1441) [reply src/main/java/com/example/demo/controller/ClientController.java]: in each if and for

File: src/main/java/com/example/demo/job/JobService.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    private final JobLauncher jobLauncher;
+    private final Job job;
+
+    public JobService(JobLauncher jobLauncher, Job job) {
+        this.jobLauncher = jobLauncher;
+        this.job = job;
+    }
+
+    private static final Logger log = Logger.getLogger(JobService.class.getName());
```
Comment (varya1441) [inline src/main/java/com/example/demo/job/JobService.java]: better to put static final before constructor

File: src/main/java/com/example/demo/job/StepProcessor.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    public StepProcessor(EventService eventService, ClientService clientService, EventMapper eventMapper, ClientMapper clientMapper) {
+        this.eventService = eventService;
+        this.clientService = clientService;
+        this.eventMapper = eventMapper;
+        this.clientMapper = clientMapper;
+    }
+
+    @Override
+    public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
```
Comment (varya1441) [inline src/main/java/com/example/demo/job/StepProcessor.java:36]: do you use it somewhere?

File: src/main/java/com/example/demo/repository/ClientRepository.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.example.demo.repository;
+
+import com.example.demo.entitie.Client;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ClientRepository extends JpaRepository<Client, Integer> {
+
+
+    //List<Client> getAllClientsByPageAndSize(Pageable pageable);
```
Comment (varya1441) [inline src/main/java/com/example/demo/repository/ClientRepository.java]: remove useless comments

File: src/main/java/com/example/demo/repository/CoachRepository.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.example.demo.repository;
+
+import com.example.demo.entitie.Coach;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface CoachRepository extends JpaRepository<Coach, Integer> {
+
+
```
Comment (varya1441) [inline src/main/java/com/example/demo/repository/CoachRepository.java]: too much empty lines

File: src/main/java/com/example/demo/repository/EventRepository.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+    List<Event> findByFitnessRoomId(Integer fitnessRoomId);
+
+    @Query(value = "SELECT * FROM event WHERE date_time > :today AND coach_id = :coachId", nativeQuery = true)
+    List<Event> getAllFutureEventsByCoach(@Param("today") OffsetDateTime today, @Param("coachId") Integer coachId);
+
+    @Query(value = "SELECT * FROM event JOIN client_event ON id = event_id WHERE client_id = :clientId", nativeQuery = true)
+    List<Event> findByClient(@Param("clientId") Integer clientId);
+
+//    @Query(value = "SELECT * FROM event WHERE date_time > :today AND client_id = :clientId", nativeQuery = true)
```
Comment (varya1441) [inline src/main/java/com/example/demo/repository/EventRepository.java]: remove useless comments

File: src/main/java/com/example/demo/service/impl/ClientServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    private final ClientMapper clientMapper;
+    private final ClientRepository clientRepository;
+    private final EventRepository eventRepository;
+
+    public ClientServiceImpl(ClientMapper clientMapper, ClientRepository clientRepository, EventRepository eventRepository) {
+        this.clientMapper = clientMapper;
+        this.clientRepository = clientRepository;
+        this.eventRepository = eventRepository;
+    }
+
```
Comment (varya1441) [inline src/main/java/com/example/demo/service/impl/ClientServiceImpl.java:30]: remove line

File: src/main/java/com/example/demo/service/impl/ClientServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+    @Override
+    public ClientDTO save(ClientDTO clientDTO) {
+        Client client = clientMapper.mapToEntity(clientDTO);
+//        List<Event> events = new ArrayList<>();
```
Comment (varya1441) [inline src/main/java/com/example/demo/service/impl/ClientServiceImpl.java]: remove useless comments

File: src/main/java/com/example/demo/service/impl/ClientServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+            event.addClient(client);
+            clientRepository.save(client);
+        }
+        else log.info("the place is full");
+    }
+
+    @Override
+    @Scheduled(fixedRate = 60000)
+    public void payment() {
+
```
Comment (varya1441) [inline src/main/java/com/example/demo/service/impl/ClientServiceImpl.java]: empty method. do you need this?

File: src/main/java/com/example/demo/service/impl/CoachServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    private final CoachMapper coachMapper;
+    private final EventRepository eventRepository;
+    private final EventMapper eventMapper;
+
+    CoachSalary salary = (coach) -> {
+        List<Event> events = coach.getEvents();
+        List<Double> costs = events.stream().map(event -> event.getFitnessRoom().getEventCost()).toList();
+        Double sum = 0.0;
+        for (Double cost : costs)
+            sum += cost;
```
Comment (varya1441) [inline src/main/java/com/example/demo/service/impl/CoachServiceImpl.java:37]: add brackets
code is not readable

File: src/main/java/com/example/demo/service/impl/EventServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+        Integer coachId = coach.getId();
+        List<Event> futureEvents = eventRepository.getAllFutureEventsByCoach(today, coachId);
+        return futureEvents.stream().map(eventMapper::mapToDTO).toList();
+    }
+
+    @Override
+    public List<EventDTO> getByClient(Integer clientId) {
+        List<Event> events = eventRepository.findByClient(clientId);
+        return events.stream().map(eventMapper::mapToDTO).toList();
+
```
Comment (varya1441) [inline src/main/java/com/example/demo/service/impl/EventServiceImpl.java]: remove line

**Review by varya1441 — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/job/StepProcessor.java (hunk 4 above)
Comment (ochenmudro) [reply src/main/java/com/example/demo/job/StepProcessor.java:36]: i think it uses when the step is built automatically

**Review by varya1441 — APPROVED** [review]

**Pull request merged**